internal/kafka/internal/workers/cluster_mgrs/deprovisioning_clusters_mgr.go

```diff
@@ -0,0 +1,128 @@
+package cluster_mgrs
+
+import (
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/config"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/services"
+	fleeterrors "github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/errors"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/workers"
+	"github.com/google/uuid"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/api"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/metrics"
+	"github.com/golang/glog"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	deprovisioningClustersWorkerType = "deprovisioning_clusters"
+)
+
+// DeprovisioningClustersManager represents a cluster manager that periodically reconciles data plane clusters in deprovisioning state.
+type DeprovisioningClustersManager struct {
+	workers.BaseWorker
+	clusterService         services.ClusterService
+	dataplaneClusterConfig *config.DataplaneClusterConfig
+}
+
+// NewDeprovisioningClustersManager creates a new cluster manager to reconcile data plane clusters in deprovisioning state.
+func NewDeprovisioningClustersManager(reconciler workers.Reconciler, clusterService services.ClusterService, dataplaneClusterConfig *config.DataplaneClusterConfig) *DeprovisioningClustersManager {
+	return &DeprovisioningClustersManager{
+		BaseWorker: workers.BaseWorker{
+			Id:         uuid.New().String(),
+			WorkerType: deprovisioningClustersWorkerType,
+			Reconciler: reconciler,
+		},
+		clusterService:         clusterService,
+		dataplaneClusterConfig: dataplaneClusterConfig,
+	}
+}
+
+// Start initializes the cluster manager to reconcile data plane clusters in deprovisioning state.
+func (m *DeprovisioningClustersManager) Start() {
+	m.StartWorker(m)
+}
+
+// Stop causes the process for reconciling data plane clusters in deprovisioning state to stop.
+func (m *DeprovisioningClustersManager) Stop() {
+	m.StopWorker(m)
+}
+
+func (m *DeprovisioningClustersManager) Reconcile() []error {
+	glog.Infoln("reconciling clusters")
+
+	var errList fleeterrors.ErrorList
+	err := m.processDeprovisioningClusters()
+	if err != nil {
+		errList.AddErrors(err)
+	}
```

Comment:
We always do this check. Maybe we could make `AddErrors` nil safe and make it no-op when the received error is `nil` so that we don't need to make this test anymore. WDYT?

Reply:
I do not know, no opinion either way. What if (hypothetically) someone wants to append `nil` in the list? 
I'll keep it like this for now. At the same time,  I'll tag @miguelsorianod in the discussion as well. I am happy for the evaluation to be done of whether we want to do the above suggestion and we could change it all over in a separate PR

Reply:
It is a list of errors, so I believe appending a `nil` would make no sense.

Reply:
> It is a list of errors, so I believe appending a nil would make no sense.

Seems like we want to make it `The list of non nil errors` because  `nil` is also an `error` ;-)